xds/src/test/java/io/grpc/xds/WeightedRoundRobinLoadBalancerTest.java

```diff
@@ -282,25 +284,47 @@ public void pickByWeight() {
     WrrSubchannel weightedSubchannel1 = (WrrSubchannel) weightedPicker.getList().get(0);
     WrrSubchannel weightedSubchannel2 = (WrrSubchannel) weightedPicker.getList().get(1);
     WrrSubchannel weightedSubchannel3 = (WrrSubchannel) weightedPicker.getList().get(2);
-    weightedSubchannel1.onLoadReport(InternalCallMetricRecorder.createMetricReport(
-            0.12, 0.1, 22, new HashMap<>(), new HashMap<>()));
-    weightedSubchannel2.onLoadReport(InternalCallMetricRecorder.createMetricReport(
-            0.28, 0.1, 40, new HashMap<>(), new HashMap<>()));
-    weightedSubchannel3.onLoadReport(InternalCallMetricRecorder.createMetricReport(
-            0.86, 0.1, 100, new HashMap<>(), new HashMap<>()));
+    weightedSubchannel1.onLoadReport(r1);
+    weightedSubchannel2.onLoadReport(r2);
+    weightedSubchannel3.onLoadReport(r3);
     assertThat(fakeClock.forwardTime(11, TimeUnit.SECONDS)).isEqualTo(1);
     Map<Subchannel, Integer> pickCount = new HashMap<>();
-    for (int i = 0; i < 1000; i++) {
+    for (int i = 0; i < 10000; i++) {
       Subchannel result = weightedPicker.pickSubchannel(mockArgs).getSubchannel();
       pickCount.put(result, pickCount.getOrDefault(result, 0) + 1);
     }
     assertThat(pickCount.size()).isEqualTo(3);
-    assertThat(Math.abs(pickCount.get(weightedSubchannel1) / 1000.0
-            - 22 / 0.12 / (22 / 0.12 + 40 / 0.28 + 100 / 0.86))).isAtMost(EDF_PRECISE);
-    assertThat(Math.abs(pickCount.get(weightedSubchannel2) / 1000.0
-            - 40 / 0.28 / (22 / 0.12 + 40 / 0.28 + 100 / 0.86) )).isAtMost(EDF_PRECISE);
-    assertThat(Math.abs(pickCount.get(weightedSubchannel3) / 1000.0
-            - 100 / 0.86 / ( 22 / 0.12 + 40 / 0.28 + 100 / 0.86) )).isAtMost(EDF_PRECISE);
+    assertThat(Math.abs(pickCount.get(weightedSubchannel1) / 10000.0 - p1)).isAtMost(EDF_PRECISE);
```

Comment:
By using subtraction, you are effectively checking that the number is within 100 of the expected value.  It would be better if you divided the actual value by the expected value and then had a bound on the ratio.
Currently, since there is no randomness, everything should be very close to the expected value.